airsonic-main/src/main/java/org/airsonic/player/dao/PodcastDao.java

```diff
@@ -157,6 +157,11 @@ public PodcastEpisode getEpisodeByUrl(String url) {
         return queryOne(sql, episodeRowMapper, url);
     }
 
+    public PodcastEpisode getEpisodeByPath(String path) {
+        String sql = "select " + EPISODE_QUERY_COLUMNS + " from podcast_episode where path=?";
```

Comment:
Unfortunately I couldn't fine any better relation between podcast episode and the media files in the db. Only thing I found was the path...

Reply:
In conclusion, there is a possibility of bad code. There is a reason that the design related to podcast has changed a little.
Media files (local) and radio (external) are relatively safe when it comes to resource integrity to paths. When considering the resource characteristics of podcasts, they have an intermediate property between local and external. (Therefore the resource status is defined) Considering only simple string comparison can result in failure by not considering status. Whether or not it actually happens depends on the hardware and software, but that's what at least the design can tell us.
So I can't give an immediate answer as to what to do, but be aware that there is such a sensitive points behind.

Reply:
It totally agree, it's defenitly a week and sensitive point here!
I guess the ideal option (from my current knowledge) would be to remove the path in the podcast episodeds table and use an id which is pointing to the media instead, when a podcast download has been finished. Path in podcast episodes seems to me kind of redundante.

I just did some more tests:
Podcasts can have the following status: _new, downloading, completed, error, deleted and 
skipped_

To come to a scenario, where this forwarding will be used (podcast is used in now playing frame, playlist or in the player [as far as I know]), the podcast episode download has to be completed. Otherwise you're not able to play the podcast and therfore can't have it in now playing, player and playlist. So I don't see an issue with _new, downloading, error and skipped_ for now.
The _deleted_ status only seems to change the status in the podcast episode table and don't doing anything else. No files are getting removed, no cleanup. So this was fine as well. I got redirected to the podcast channel showing all available episodes (even though the selected was not showing there anymore). 
But when deleting a whole podcast channel I was able to produce a null pointer exception: When adding a podcast to a playlist and removing the whole podcast channel, the podcast episode is still in the playlist. The playlist already stated the file to be missing. By clicking on the podcast channel name there in the playlist you'll get a null pointer exception. Should I try to catch this? If yes, what to do in this case? I'm not quite sure what's the best option here.

Reply:
"I'm not quite sure what's the best option here" should be one of the correct answers at this point. 

I think most podcasts process can only be implemented with callbacks. Presence check and status check are required.
If it is not asynchronous communication, it will be an implementation that performs manual update on the assumption that the process fails.


The current web implementation is simple and should be considered a draft implementation. Legacy implementations may not be very informative. (the current implementation is not implemented enough high-quality   helpful for collecting information.)

___

 > Legacy implementations may not be very informative.

It's off story, I think there are some features in Airsonic that are more desirable to create than fix. I'm talking with fxthomas, it is desirable to create a new detailed search pages.
To be honest, I also think podcasts should have a zero start. I'm not bluntly denying the original implementation. Because there is also one side that it was difficult with the technology at the time of implementation. I imagine that we could make something better, given the current new asynchronous communication technology.

I'm thinking of that.
I'm not against what this PR is trying to achieve. If anything, I think it's part of the improvement needed.